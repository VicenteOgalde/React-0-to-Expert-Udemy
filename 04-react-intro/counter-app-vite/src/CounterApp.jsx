import { useState } from 'react';
import PropTypes from 'prop-types';




export const CounterApp=({ counter })=>{

    const [value,setValue]=useState(counter);

    const handleAdd=()=>{
        setValue((c)=>c+1);
    }
    return <>
    <h1>CounterApp</h1>
    <h2>{ value }</h2>
    <button onClick={handleAdd}>+1</button>
    </>
}

CounterApp.propTypes={
    counter:PropTypes.number
}
CounterApp.defaultProps={
    counter:0
}