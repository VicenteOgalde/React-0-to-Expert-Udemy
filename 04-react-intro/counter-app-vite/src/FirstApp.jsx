export function FirstApp(){
    return <h2>First App</h2>
}
