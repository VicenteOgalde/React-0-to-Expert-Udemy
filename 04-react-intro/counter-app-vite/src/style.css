html,body{
    background-color: cadetblue;
    color: brown;
    font-family: sans-serif,Arial, Helvetica;
    font-size: 1.3rem;
}