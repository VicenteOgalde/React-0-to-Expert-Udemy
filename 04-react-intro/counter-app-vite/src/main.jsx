import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp';
import {App} from './HelloWorldApp'
import './style.css'

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <App></App>
        <FirstApp title='nice App' subTitle='very nice App'></FirstApp>
    </React.StrictMode>
)