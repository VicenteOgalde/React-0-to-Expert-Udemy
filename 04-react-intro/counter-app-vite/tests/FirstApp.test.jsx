/** * @jest-environment jsdom */
import { getByText, render } from '@testing-library/react';
import { FirstApp } from './../src/FirstApp';

describe('test FirsApp', () => {

     test('should make match with the snapshot', () => {

        const {container}=render(<FirstApp title='nice App' subTitle='very nice App'></FirstApp>)

        expect(container).toMatchSnapshot();

         });

         test('should show the title in a H1', () => {

          const title ='good morning'
          const {container, getByText}=render(<FirstApp title={title} subTitle='very nice App'></FirstApp>)
  
          expect(getByText(title) ).toBeTruthy();
  
           });  
    });