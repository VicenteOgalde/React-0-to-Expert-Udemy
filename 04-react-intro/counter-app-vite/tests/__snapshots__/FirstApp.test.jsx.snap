// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test FirsApp should make match with the snapshot 1`] = `
<div>
  <h2>
    First App 
    hello2
  </h2>
  <h3>
    nice App
  </h3>
  <h3>
    very nice App
  </h3>
</div>
`;

exports[`test FirsApp should make match with the snapshot 2`] = `
<div>
  <h2>
    First App 
    hello2
  </h2>
  <h3>
    nice App
  </h3>
  <h3>
    very nice App
  </h3>
</div>
`;

exports[`test FirsApp should show the title  1`] = `
<div>
  <h2>
    First App 
    hello2
  </h2>
  <h3>
    nice App
  </h3>
  <h3>
    very nice App
  </h3>
</div>
`;

exports[`test FirsApp should show the title in a H1 1`] = `
<div>
  <h2>
    First App 
    hello2
  </h2>
  <h3>
    nice App
  </h3>
  <h3>
    very nice App
  </h3>
</div>
`;
